homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -142,19 +135,29 @@ def device_state_attributes(self):
             _LOGGER.debug("Got new state: %s", state)
 
             self._available = True
-            self._state = state.aqi
-            self._state_attrs.update(
-                {
-                    ATTR_POWER: state.power,
-                    ATTR_CHARGING: state.usb_power,
-                    ATTR_BATTERY_LEVEL: state.battery,
-                    ATTR_DISPLAY_CLOCK: state.display_clock,
-                    ATTR_NIGHT_MODE: state.night_mode,
-                    ATTR_NIGHT_TIME_BEGIN: state.night_time_begin,
-                    ATTR_NIGHT_TIME_END: state.night_time_end,
-                    ATTR_SENSOR_STATE: state.sensor_state,
-                }
-            )
+            if self._model != MODEL_AIRQUALITYMONITOR_B1:
```

Comment:
This looks different enough that it warrants a new class

Reply:
I agree the devices are quite different. But in the python-miio airqualitymonitor class the properties are already defined for each device. So I suggest we'd always go for the generic approach here, like:

`self._state = state.temperature`
`self._unit_of_measurement = '°C'`
`self._state_attrs.update(state.data)`

Where unit of measurement can be defined in the python-miio class.

Reply:
Did you the my PR for the S1? It must be rewritten to use the Air Quality Entity but you could adopt the structure. Furthermore you could derive from the XiaomiAirQualityMonitor class and override some methods.